includes/theme.inc

```diff
@@ -1675,7 +1675,7 @@ function theme_blocks($region) {
  */
 function theme_username($object) {
 
-  if ($object->uid && $object->name) {
+  if (isset($object->uid) && isset($object->name)) {
```

Comment:
In order to change as little as possible (which seems significant for a codebase like D6), all 3 `isset()` should be '!empty()'. Otherwise this first code block will start being used, when it wasn't before... if you pass the anonymous user into it.